### PR TITLE
Chat highlighting when username is mentioned

### PR DESCRIPTION
**Pull request by poixen**
**Features**
+ When you mentioned in chat (@username) you will see the entry highlighted
orange mention

+ Turned on by default, but can be disabled in the options menu.
chatsettings

### CODE REVIEW

**Comment by ZeldaZach** [comment]
Is considering just highlighting the username in blue an option instead of the entire line in red?

**Comment by Nightfirecat** [comment]
Just my two cents, but I'd likely prefer a chat line have its background highlighted (moderate yellow color, I'd imagine) rather than have a full line of chat turn red or blue - keeps the text more readable.

**Comment by ZeldaZach** [comment]
I like the background color idea @Nightfirecat

File: cockatrice/src/chatview.cpp [hunk 1]
```diff
@@ -168,8 +168,15 @@ void ChatView::appendMessage(QString message, QString sender, UserLevelFlags use
         } else
             from = 1;
     }
+
+    QTextCharFormat charFormat;
+    if (message.contains(QString::fromStdString(tabSupervisor->getUserInfo()->name()))) {
```
Comment (Daenyth) [inline cockatrice/src/chatview.cpp]: Maybe cache name? It's not likely to change.

File: cockatrice/src/chatview.cpp [hunk 2]
```diff
@@ -168,8 +168,15 @@ void ChatView::appendMessage(QString message, QString sender, UserLevelFlags use
...
+        charFormat.setFontWeight(QFont::Bold);
+        charFormat.setForeground(Qt::red);
```
Comment (Daenyth) [inline cockatrice/src/chatview.cpp]: Can this be done with a css class and the style decoupled so it's easier to change?

**Comment by ZeldaZach** [comment]
Additional ideas:

Twitch.tv has a 3rd party program, BetterTTV
I can enter in a field "HIGHLIGHT THE LINE FOR PHRASE(S)" and it'll highlight all lines that have any of the phrases
screenshot 2015-01-08 22 20 08

**Comment by Daenyth** [comment]
This PR partially addresses #458

File: cockatrice/src/chatview.cpp (hunk 1 above)
Comment (Daenyth) [inline cockatrice/src/chatview.cpp]: Case insensitive compare makes sense too.

**Comment by tooomm** [comment]
Can you make the color configurable in settings to fit every needs?
But red as default is ok I guess.

An Option to use Bold, Italic etc. too? Or a mixture?
Just going with black bold or black italic e.g. still provides enough "highlighting" and doesn't hurt your eyes. :)

I think I like background highlighting of the word most, but that's not an option right now.

**Comment by Daenyth** [comment]
I haven't tested it but google says this http://colororacle.org/

**Comment by poixen** [comment]
Tested with the software, you are able to determine when you have been mentioned.

**Comment by poixen** [comment]
Been thinking about this and the following will be added:

**Common usernames**
If you have a username like "what" then this feature will be hell for you, therefor I am thinking the "@" must precede the mention. That way if you do have a very common user name, then it will only alert when "@" is used.

**Disable in the settings**
I am adding the ability to enable/disable in the settings

**Configure through the settings**
I am going to allow you to configure the color through the settings.

**Comment by woogerboy21** [comment]
Yeah, im not big into the sharpness of the red and another color would be nice.  Self defined would be great but if that isn't going to be possible than a much more smoother toned would be nice.

**Comment by poixen** [comment]
How do we like this? Its the same color as the new stack zone color?

chat

File: cockatrice/src/chatview.cpp [hunk 3]
```diff
@@ -168,8 +170,17 @@ void ChatView::appendMessage(QString message, QString sender, UserLevelFlags use
         } else
             from = 1;
     }
+
+    QTextCharFormat charFormat;
+    if (settingsCache->getChatMention()) {
+        if (message.contains("@" + QString::fromStdString(tabSupervisor->getUserInfo()->name()))) {
```
Comment (Daenyth) [inline cockatrice/src/chatview.cpp]: Case insensitive compare?

Comment (poixen) [inline cockatrice/src/chatview.cpp]: I tried adding Qt::CaseInsensative but it didn't work. Not sure what to do.

Comment (ZeldaZach) [inline cockatrice/src/chatview.cpp]: Is there a `string_to_lower()` option?

Comment (Daenyth) [inline cockatrice/src/chatview.cpp]: If message.to lower.contains(name.tolower)

**Comment by poixen** [comment]
Updated to support case insensitive mentions.

**Comment by ZeldaZach** [comment]
Looks fine to me, but I'll have to test to be certain.

**Comment by poixen** [comment]
I need to change the color, when using normal chat its hard to see the color.

**Comment by poixen** [comment]
I have updated the color from a hard rest to a softer orange.

**Comment by Daenyth** [comment]
I think this is just about ready to go

**Comment by ZeldaZach** [comment]
I think we're good

**Comment by Daenyth** [comment]
@poixen merge when ready after travis passes

**Comment by poixen** [comment]
Will do. Again this is the first step in ~3 part phase to get your own custom chat settings.

**Comment by ZeldaZach** [comment]
Glad you're making progress. I'm sure you'll figure out the rest whenever you're ready x)

**Comment by tooomm** [comment]
Maybe the confusing part is that you username is always the same color than the chat line you being mentioned in?
What about a color for your own chats (username) and another color for chats related to you (whole line)?

But it could just be the picture, because you tested it for yourself with no one interacting with you.

Someone gave this a run with normal chat going on?

**Comment by poixen** [comment]
@tooomm Its the same color intentionally. I want users to correlate a color (orange now, but in the future will be user set) to them-self. That includes mentions and seeing your own user name. Hope that makes sense.

**Comment by tooomm** [comment]
I'm happy to test it in the next weekly build! Need to experience it under normal conditions. :)

btw: Looking forward to your other 66% of pure chat power. Hehe.

**Pull request merged**